### ISSUE TITLE
`options` collection of a Node should be accessible as a Map in Drab.Element

### ISSUE THREAD

**Issue by mbenatti** [grych/drab#75]
Hello, 
I have an select with `drab-event='change'`, sometimes I have experiencing this <select> being disabled infinity (after 4/5 requests I got <select> disabled>)

The select in #select1 will call coin_selected in Drab commander and change html contents of #select2
```HTML
<div class="row marketing jumbotron">
  <div class="col-lg-12">

    <div id="select1">
      <label>Select some option:</label>
      <select drab-event='change' drab-handler='coin_selected' class="selectpicker" data-live-search="true">
        <option>USD</option>
        <option>BRL</option>
        <option>BTC</option>
      </select>
    </div>

    <div id="select2">
      <select class="selectpicker" data-live-search="true">
        <option>OPT 1</option>
        <option>OPT 2</option>
        <option>OPT 3</option>
      </select>
    </div>
  </div>

</div>
```

My drab commander:
```Elixir
  def coin_selected(socket, _sender) do
    data = [%{name: "BITCOIN", symbol: "BTC"},
            %{name: "LITECOIN", symbol: "LTC"},
            %{name: "MONERO", symbol: "XMR"},
            %{name: "BITCOIN GOLD", symbol: "BTG"}]

    html = ~E"""
          <select class="selectpicker" data-live-search="true">
            <%= for map <- data do %>
              <option><%= map.name %> - <%= map.symbol %></option>
            <%= end %>
          </select>
          """
          |> safe_to_string()

    set_prop socket, "#select2", innerHTML: html
    socket |> exec_js("$('.selectpicker').selectpicker();")
  end
```

What happens:

![hello testdrab](https://user-images.githubusercontent.com/3832221/35480085-3cb4496c-03ee-11e8-8514-66e257e3a114.png)

**Comment by grych** [comment]
HI @mbenatti,
Did you set `:events_to_disable_while_processing` config value by any chance?

If not, this is not Drab disabling your `select`, Drab disables only elements having `click` event by default. I suspect the `selectpicker`, but not sure. 

I checked it with vanilla html, the issue is not reproducible.

BTW maybe it would be easier not to destroy, re-create and re-issue `selectpicker()` on the second select, update only the `options` list? You would not need to run this selectpicker every time.

**Comment by mbenatti** [comment]
Hey @grych 

Thanks for the quick response, 
You are right, I will do some tests debugging this `selectpicker()`

**Comment by mbenatti** [comment]
@grych How can I update only the options list?
In your examples I found only functions changing `div` contents and not for other elements

**Comment by grych** [comment]
There are few different ways to do it in the html. The easiest is based on the observation, that options list is just an `innerHTML` of the existing select, so you may do something like (notice the selector, do it better):

    html = ~E"""
            <%= for map <- data do %>
              <option><%= map.name %> - <%= map.symbol %></option>
            <% end %>
          """
          |> safe_to_string()

    set_prop socket, "select:nth-child(1)", innerHTML: html

The other option is to build select option collection, see https://www.w3schools.com/jsref/coll_select_options.asp

**Comment by grych** [comment]
Hmm it would be good if you could manipulate `options` directly from Drab, as a good old map, eg:

```elixir
    set_prop, socket, "#select_1", options: %{"value 1" => "key 1"}
```

Reopening the issue.

**Comment by mbenatti** [comment]
@grych Amazing, I'll test it soon

**Comment by grych** [comment]
The change will be included in 0.7.1,  or you may try it now from master. Enjoy!
